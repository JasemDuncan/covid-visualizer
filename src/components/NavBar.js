import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faChevronLeft,
  faMicrophone, faCog, faSignal,
} from '@fortawesome/free-solid-svg-icons';

const NavBar = () => (
  <div className="row d-flex justify-content-center align-items-center mt-3">
    <div className="col-2 d-flex justify-content-center align-items-center">
      <NavLink to="/">
        <FontAwesomeIcon icon={faChevronLeft} size="sm" color="white" />
      </NavLink>
    </div>
    <div className="col-7 d-flex justify-content-center align-items-center">
      <div className="title">COVID VISUALIZER â„¢</div>
    </div>
    <div className="col-3 d-flex justify-content-around align-items-center">
      <FontAwesomeIcon icon={faSignal} size="sm" color="yellow" />
      <FontAwesomeIcon icon={faMicrophone} size="sm" color="white" />
      <FontAwesomeIcon icon={faCog} size="sm" color="white" />
    </div>
  </div>

);

export default NavBar;
