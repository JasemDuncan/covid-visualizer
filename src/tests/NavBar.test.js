import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter, NavLink } from 'react-router-dom';
import NavBar from '../components/NavBar';

describe('Test for NavBar', () => {
  test('Should match to snapshot', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    ).toJSON;
    expect(tree).toMatchSnapshot();
  });

  test('Should render NavBar component', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    ).root;
    expect(tree.findByType(NavLink).props.to).toBe('/');
  });

  test('should match the label text', () => {
    const testRenderer = renderer.create(
      <MemoryRouter><NavBar /></MemoryRouter>,
    ).root;
    expect(testRenderer.findByProps({ className: 'title' }).children).toEqual(['COVID VISUALIZER â„¢']);
  });
});
