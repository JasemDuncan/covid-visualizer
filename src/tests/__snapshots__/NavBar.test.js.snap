// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for NavBar Should match to snapshot 1`] = `[Function]`;
